File: services/typescript/express/src/routes/handlers/invalid_path.ts
import type { RequestHandler } from "express";
import { AppErrors, sendError } from "../errors.js";
import { getLogger } from "telemetry";

export const invalidPathHandler: RequestHandler = (req, res) => {
  const logger = getLogger();
  logger.warn({ method: req.method, path: req.path }, "Invalid path");
  sendError(res, AppErrors.InvalidPath);
};

File: services/python/django/src/routes/handlers/invalid_path.py
import structlog
from django.http import HttpRequest, JsonResponse

from routes.errors import AppErrors, send_error


def invalid_path_handler(request: HttpRequest, exception: Exception) -> JsonResponse:
    structlog.get_logger().warn(
        "Invalid path", method=request.method, path=request.path
    )
    return send_error(AppErrors.InvalidPath)
